README.md

```diff
@@ -116,6 +116,14 @@ To check whether dnsmasq name server at 127.0.0.1 can resolve subdomains of dev.
 app.dev.gov.uk.		0	IN	A	127.0.0.1
 ```
 
+## Environment variables
+
+Both govuk-docker and the Makefile respect the following environment variables:
+
+- `$GOVUK_ROOT_DIR` - directory where app repositories live, defaults to `$HOME/govuk`
+- `$GOVUK_DOCKER_DIR` - directory where the govuk-docker repository lives, defaults to `$GOVUK_ROOT_DIR/govuk-docker`
```

Comment:
I'm not sure this one is actually respected, since we do a '..' in all the compose files to mount the govuk directory. Perhaps we should change that?

Reply:
That's a good point.  I'd be in favour of changing that - can you lookup environment variables with a default value in compose files?

Reply:
https://docs.docker.com/compose/environment-variables/ doesn't have anything on it.  So I think if we want to use `$GOVUK_ROOT_DIR` in the compose files we'd have to have `govuk-docker` set it before calling `docker-compose`.  Does that sound reasonable, or is it too magic?

Reply:
That's annoying. Ideally a user should be able to copy the verbose command and achieve the same result, so we maintain some transparency - it's been useful when I've debugged stuff. Could it a prefix to `docker-compose` i.e. `GOVUK_ROOT_DIR=... docker-compose`?

Reply:
I'll try things out in another branch

Reply:
Ah, I was looking at the wrong bit of the manual.  This is the relevant bit: https://docs.docker.com/compose/compose-file/#variable-substitution

> `${VARIABLE:-default}` evaluates to `default` if `VARIABLE` is unset or empty in the environment.

Reply:
docker-compose doesn't seem to like when the default value contains an environment variable.  These don't work:

```
${GOVUK_ROOT_DIR-$HOME/govuk}:/govuk:delegated
${GOVUK_ROOT_DIR-"$HOME/govuk"}:/govuk:delegated
${GOVUK_ROOT_DIR-"$HOME"/govuk}:/govuk:delegated
${GOVUK_ROOT_DIR-${HOME}/govuk}:/govuk:delegated
${GOVUK_ROOT_DIR-"${HOME}/govuk"}:/govuk:delegated
${GOVUK_ROOT_DIR-"${HOME}"/govuk}:/govuk:delegated
```

Just to confirm that the default syntax works, this *does*:

```
${GOVUK_ROOT_DIR-HOME/govuk}:/govuk:delegated
```

Argh